lib/oppen/scan_stack.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# Oppen
+module Oppen
+  # Stack used by the scan in the Oppen algorithm
+  class ScanStack
+    def initialize(size)
+      @stack = Array.new(size)
+      @empty = true
+      @top = @bottom = 0
```

Comment:
make all of these actual accessors. They're easier to read and type :)

Reply:
Doing so would create a confilct between the attributes top/bottom and the methods top/bottom